src/hotspot/share/jeandle/jeandleCompilation.hpp

```diff
@@ -36,6 +36,42 @@
 #include "ci/ciMethod.hpp"
 #include "memory/arena.hpp"
 
+#include "runtime/timer.hpp"
+#include "runtime/timerTrace.hpp"
+#include "compiler/compiler_globals.hpp"
+
+enum JeandleTimerName : int {
+  Compile = 0,           // Total compilation time
+  AbstractInterpreter,   // Jeandle abstract interpreter phase
+  LlvmOptimizer,         // LLVM optimization phase
+  LlvmCodegen,           // LLVM code generation phase
+  Finalize,              // JeandleCompiledCode::finalize phase
+  MaxPhase               // Number of Jeandle timer phases (sentinel)
+};
+
+// Static timer array, corresponding to C1's Compilation::timers[]
+static elapsedTimer jeandle_timers[MaxPhase];
+
+// Counts how many methods have been compiled by Jeandle (optional)
+static int jeandle_compilation_count = 0;
+
+class JeandleTraceTime : public TraceTime {
+ private:
+  JeandleTimerName _timer;
+
+ public:
+  JeandleTraceTime(const char* name, JeandleTimerName timer_name)
+  : TraceTime(name,
+              &jeandle_timers[timer_name],
+              CITime || CITimeEach,
+              CITimeEach && CITimeVerbose),
```

Comment:
```suggestion
              CITimeVerbose),
```
I think the condition `CITimeEach && CITimeVerbose` is ambiguous. Only CITimeVerbose is clearer description.